packages/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -237,13 +259,25 @@ class _GoogleMapOptions {
     addIfNonNull('zoomGesturesEnabled', zoomGesturesEnabled);
     addIfNonNull('trackCameraPosition', trackCameraPosition);
     addIfNonNull('myLocationEnabled', myLocationEnabled);
+    addIfNonNull('markers', _markersToJson());
+
     return optionsMap;
   }
 
-  Map<String, dynamic> updatesMap(_GoogleMapOptions newOptions) {
+  dynamic _markersToJson() {
+    if (markers == null) {
+      return null;
+    }
+    return markers.map<dynamic>((Marker marker) => marker._toJson())?.toList();
+  }
+
+  Map<String, dynamic> updatesMap(
+      _GoogleMapOptions newOptions, MarkerUpdates markerUpdates) {
     final Map<String, dynamic> prevOptionsMap = toMap();
+
     return newOptions.toMap()
-      ..removeWhere(
-          (String key, dynamic value) => prevOptionsMap[key] == value);
+      ..remove('markers')
+      ..removeWhere((String key, dynamic value) => prevOptionsMap[key] == value)
+      ..putIfAbsent('markerUpdates', () => markerUpdates._toJson());
```

Comment:
Took me a little bit to wrap my ahead around who's responsible for the marker updates.

It used to be that `_GoogleMapOptions` is holds all the options data, this data is passed in the creation parameters 'options' field, `_GoogleMapOptions` knows to compute update deltas which are passed to `_updateMapOptions`.

The `markers` field doesn't follow this same pattern: it is still passed as part of the options in the creation parameters, but _GoogleMapOptions is not responsible for computing updates for it, and `_updateMapOptions` ignores it.

What do you think about keeping a _markers field in _GoogleMapState?